crypto/x509/by_store.c

```diff
@@ -115,27 +135,54 @@ static int by_store_ctrl_ex(X509_LOOKUP *ctx, int cmd, const char *argp,
     switch (cmd) {
     case X509_L_ADD_STORE:
         if (argp != NULL) {
-            STACK_OF(OPENSSL_STRING) *uris = X509_LOOKUP_get_method_data(ctx);
-            char *data = OPENSSL_strdup(argp);
+            STACK_OF(CACHED_STORE) *stores = X509_LOOKUP_get_method_data(ctx);
+            CACHED_STORE *data = OPENSSL_malloc(sizeof(*data));
 
             if (data == NULL) {
                 return 0;
             }
-            if (uris == NULL) {
-                uris = sk_OPENSSL_STRING_new_null();
-                X509_LOOKUP_set_method_data(ctx, uris);
+            memset(data, 0, sizeof(*data));
```

Comment:
Why not just use zalloc in the first place?

Reply:
Oh, 'cause I forget it exists...  fixing.